requests/utils.py

```diff
@@ -116,7 +116,10 @@ def get_netrc_auth(url, raise_errors=False):
         ri = urlparse(url)
 
         # Strip port numbers from netloc
-        host = ri.netloc.split(':')[0]
+        if isinstance(url, str):
```

Comment:
Suggestion:

```py
splitstr = b':'
if isinstance(url, str):
    splitstr = splitstr.decode('utf-8')
host = r.netloc.split(splitstr)[0]
```

My reasoning is that while we don't test against Python 3.2, pip still supports 3.2 and so we need to support it as best as we can as well. `u''` is not supported on 3.2 and that's why the above might be a bit friendlier.